### PR TITLE
use _data instead of data in MutableSeq to be consistent with Seq

### PR DESCRIPTION
**Pull request by mdehoon**
This pull request renames the `data` attribute of `MutableSeq` to `_data` to be consistent with the `Seq` class.

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``pre-commit`` locally,
and understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

**Comment by peterjc** [comment]
I don't think this is a good idea. It is a backward compatibility breaking change, likely to cause trouble for any serious users of the ``MutableSeq`` (a minority to be sure). At a minimum it needs flagging in the DEPRECATED file, perhaps also the NEWS file.

**Comment by mdehoon** [comment]
We could add a `data` attribute getter/setter that points to the `_data` attribute, and add a `DeprecationWarning` to that.

**Comment by peterjc** [comment]
That's a good idea if you really want to kill the ``.data`` attribute.

**Comment by mdehoon** [comment]
Quoting from the `DEPRECATED` file:
```
Bio.Seq and the data property
-----------------------------
Direct use of the Seq object (and MutableSeq object) .data property is
deprecated.  As of Release 1.49, writing to the Seq object's .data property
triggered a warning, and this property was made read only in Release 1.53. In
Release 1.55 final, accessing the .data property gives a DeprecationWarning.
The Seq object's .data property was removed in Release 1.61.
```
It looks like keeping the `.data` property on `MutableSeq` was just an oversight.

**Comment by peterjc** [comment]
Yes. It does look like we should have added the warning on get/set on the ``MutableSeq`` back then too. Better late than never (although I'd be willing to deprecate the entire ``MutableSeq`` object instead?).

File: Bio/Seq.py [hunk 1]
```diff
@@ -1708,6 +1708,26 @@ def __init__(self, data):
...
+    @data.setter
+    def data(self, value):
+        warnings.warn(
+            "Accessing MutableSeq.data has been deprecated, as it is a private "
+            "attribute. Please use indexing to access the sequence contents of "
+            "a MutableSeq object.",
+            BiopythonDeprecationWarning,
+        )
+        self._data = value
+
```
Comment (peterjc) [inline Bio/Seq.py:1730]: Maybe a slight rewording, "is now a private attribute"?

File: DEPRECATED.rst [hunk 2]
```diff
@@ -427,13 +427,15 @@ Numeric support
...
+---------------------------------------------
 Direct use of the Seq object (and MutableSeq object) .data property is
 deprecated.  As of Release 1.49, writing to the Seq object's .data property
 triggered a warning, and this property was made read only in Release 1.53. In
-Release 1.55 final, accessing the .data property gives a DeprecationWarning.
-The Seq object's .data property was removed in Release 1.61.
+Release 1.55 final, accessing the .data property of a Seq object gives a
+DeprecationWarning. The Seq object's .data property was removed in Release 1.61.
+Starting from Release 1.78, accessing the .data property of a MutableSeq object
+similarly gives a DeprecationWarning.
```
Comment (peterjc) [inline DEPRECATED.rst:438]: maybe "deprecation warning" or just "warning" since it isn't literally the ``DeprecationWarning ``.

**Comment by mdehoon** [comment]
> .. (although I'd be willing to deprecate the entire `MutableSeq` object instead?).

Having a mutable sequence object is definitely useful. Just the implementation of `MutableSeq` is problematic.

File: DEPRECATED.rst (hunk 2 above)
Comment (mdehoon) [reply DEPRECATED.rst]: done

File: Bio/Seq.py (hunk 1 above)
Comment (mdehoon) [reply Bio/Seq.py:1732]: done

**Review by peterjc — APPROVED** [review]

**Pull request merged**